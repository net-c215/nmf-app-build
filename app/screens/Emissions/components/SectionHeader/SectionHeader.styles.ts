import { StyleSheet } from "react-native";
import { Colors } from "../../../../style";
import { PADDING_HORIZONTAL } from "../../../../constants/Layout";

const styles = StyleSheet.create({
  container: {
    paddingVertical: 10,
    paddingLeft: PADDING_HORIZONTAL,
    backgroundColor: Colors.white,
    borderBottomColor: Colors.gray,
    borderBottomWidth: 2,
  },
});

export default styles;
