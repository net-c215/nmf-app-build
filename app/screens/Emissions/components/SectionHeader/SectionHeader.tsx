import React from "react";
import { View } from "react-native";
import moment from "moment";
import "moment/min/locales";
import { Text } from "../../../../components";
import styles from "./SectionHeader.styles";
import { withLocalization } from "../../../../utils";

const SectionHeader = ({ date, language = "" }) => (
  <View style={styles.container}>
    <Text.Primary darkGray bold style={styles.text}>
      {moment(date).locale(language).format("MMMM YYYY")}
    </Text.Primary>
  </View>
);

export default withLocalization(SectionHeader);
