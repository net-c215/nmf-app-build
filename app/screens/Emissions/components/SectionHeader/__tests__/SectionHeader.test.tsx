import React from "react";
import { create } from "react-test-renderer";

import SectionHeader from "../SectionHeader";

jest.unmock("../SectionHeader");

describe("SectionHeader", () => {
  const date = new Date("2020-12-24T03:24:00");

  it("should render when user is over budget", () => {
    const tree = create(<SectionHeader date={date} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when user is below budget", () => {
    const tree = create(<SectionHeader date={date} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when user has budget below 1%", () => {
    const tree = create(<SectionHeader date={date} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when co2value equals 0", () => {
    const tree = create(<SectionHeader date={date} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when monthlyCarbonBudget equals 0", () => {
    const tree = create(<SectionHeader date={date} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render when monthlyCarbonBudget and co2value equals 0", () => {
    const tree = create(<SectionHeader date={date} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
