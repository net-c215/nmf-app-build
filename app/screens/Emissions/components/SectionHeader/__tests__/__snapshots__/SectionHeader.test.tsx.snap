// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader should render when co2value equals 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Text.Secondary
      blue={true}
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      More Info
    </Text.Secondary>
  </TouchableOpacity>
</View>
`;

exports[`SectionHeader should render when monthlyCarbonBudget and co2value equals 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Text.Secondary
      blue={true}
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      More Info
    </Text.Secondary>
  </TouchableOpacity>
</View>
`;

exports[`SectionHeader should render when monthlyCarbonBudget equals 0 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Text.Secondary
      blue={true}
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      More Info
    </Text.Secondary>
  </TouchableOpacity>
</View>
`;

exports[`SectionHeader should render when user has budget below 1% 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Text.Secondary
      blue={true}
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      More Info
    </Text.Secondary>
  </TouchableOpacity>
</View>
`;

exports[`SectionHeader should render when user is below budget 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Text.Secondary
      blue={true}
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      More Info
    </Text.Secondary>
  </TouchableOpacity>
</View>
`;

exports[`SectionHeader should render when user is over budget 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DCF0E5",
      "borderBottomWidth": 2,
      "elevation": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "transparent",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowRadius": 0,
    }
  }
>
  <Text.Primary
    bold={true}
    darkGray={true}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
  >
    December 2020
  </Text.Primary>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Text.Secondary
      blue={true}
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      More Info
    </Text.Secondary>
  </TouchableOpacity>
</View>
`;
